spec/order_repository_spec.rb

```diff
@@ -0,0 +1,47 @@
+require 'order_repository'
+
+RSpec.describe OrderRepository do
+  # snippet and function call copied and modified from repository class recipe
+  def reset_tables
+    seed_sql = File.read('spec/seeds_shop_manager.sql')
+    connection = PG.connect({ host: '127.0.0.1', dbname: 'shop_manager_test' })
+    connection.exec(seed_sql)
+  end
+
+  before(:each) do 
+    reset_tables
+  end
+
+  describe "#all_with_item" do
+    it "returns a list of [Order, item name string] for each order" do
+      order_repo = OrderRepository.new
+      orders = order_repo.all_with_item
+      expect(orders.length).to eq 4
+      expect(orders.first[0].id).to eq 1
+      expect(orders.first[0].customer_name).to eq "John Doe"
+      expect(orders.first[0].order_date).to eq "2022-08-21"
+      expect(orders.first[0].item_id).to eq 2
+      expect(orders.first[1]).to eq "Makerspresso Coffee Machine"
+      expect(orders.last[0].id).to eq 4
+      expect(orders.last[0].customer_name).to eq "Jane Doe"
+      expect(orders.last[0].order_date).to eq "2022-08-24"
+      expect(orders.last[0].item_id).to eq 1
+      expect(orders.last[1]).to eq "Super Shark Vacuum Cleaner"
```

Comment:
Here this` first[0]` notation is not readable. I didn't understand what was going on, why you were using array access notation when you'd gotten the last order in the orders array. I had to go back and look at your `Order` and `OrderRepository `classes to figure it out. It is the line `orders_with_item << [order, record['name']]` where you are storing arrays rather than just an order object. So the name of the item is stored as a second item in the array? Why not store it in the `@items` variable you have in your `Order `objects. Than you can access `order.last.items[0].name`. Now it is clear to me when reading that line that you are accessing the name of an item in the order!